jdaviz/helpers.py

```diff
@@ -1,50 +1,87 @@
 
-    def load_data(self, data, data_label='', format=None):
-        """Loads a data file or Spectrum1D object into SpecViz
+    def load_data(self, data, data_label=None, format=None):
+        """
+        Loads a data file or `~specutils.Spectrum1D` object into SpecViz.
 
-        :param data: Spectrum1D spectra, or path to compatible data file
-        :param data_label: Name/identifier of data
-        :param format: Spectrum1D data format to load
+        Parameters
+        ----------
+        data : str or `~specutils.Spectrum1D`
+            Spectrum1D spectra, or path to compatible data file.
+        data_label : str
+            The Glue data label found in the ``DataCollection``.
+        format : str
+            Loader format specification used to indicate data format in
+            `~specutils.Spectrum1D.read` io method.
         """
         # If no data label is assigned, give it a unique identifier
-        if not data_label:
+        if data_label is None:
             data_label = "specviz_data|" + uuid.uuid4().hex
+
         # If data provided is a path, try opening into a Spectrum1D object
         try:
             path = pathlib.Path(data)
+
             if path.is_file():
                 data = Spectrum1D.read(path, format=format)
             else:
                 raise FileNotFoundError("No such file: " + path)
         # If not, it must be a Spectrum1D object. Otherwise, it's unsupported
         except TypeError:
             if type(data) is SpectrumCollection:
-                raise TypeError("SpectrumCollection detected. Please provide a Spectrum1D")
+                raise TypeError("`SpectrumCollection` detected. Please "
+                                "provide a `Spectrum1D`.")
             elif type(data) is not Spectrum1D:
-                raise TypeError("Data is not a Spectrum1D object or compatible file")
+                raise TypeError("Data is not a Spectrum1D object or "
+                                "compatible file.")
         self._app.add_data(data, data_label)
         self._app.add_data_to_viewer('spectrum-viewer', data_label)
 
     def get_spectra(self):
         """Returns the current data loaded into the main viewer"""
         return self._app.get_data_from_viewer('spectrum-viewer')
+
+    def get_regions(self):
+        """
+        Retrieves glue subset objects from the spectrum viewer and converts
+        them to `~specutils.SpectralRegion` objects.
+
+        Returns
+        -------
+        spec_regs : dict
+            Dictionary wherein the values are the converted subsets and the
+            keys are the name reference of the subset objects.
+        """
+        regions = self._app.get_subsets_from_viewer('spectrum-viewer')
+
+        spec_regs = {}
+
+        for name, reg in regions.items():
+            spec_reg = SpectralRegion((reg.center.x - reg.width * 0.5,
+                                       reg.center.x + reg.width * 0.5))
```

Comment:
Probably better to use this construct:
```suggestion
            spec_reg = SpectralRegion.from_center(reg.center.x, reg.width)
```
which amounts to the same now but *might* be subtly different in the future.